src/vs/workbench/browser/parts/editor/editorPart.ts

```diff
@@ -624,6 +653,10 @@ export class EditorPart extends Part implements IEditorPart {
 
 	private doSetGroupActive(group: IEditorGroupView): void {
 		if (this._activeGroup !== group) {
+			if (this._activeGroup && this.isGroupMaximized(this._activeGroup)) {
+				this.unmaximizeGroup();
+			}
```

Comment:
Isnt this code doing exactly what `this.showActiveEditorGroupsIfHidden();` is doing below now? If so, then I would just remove `showActiveEditorGroupsIfHidden` method.